src/components/List/List.js

```diff
@@ -0,0 +1,162 @@
+        }
+    });
+
+
+    //FUNCTIONS-----------------------------------------------------------------
+    const newButtonClickHandler = () => {
+        set(ref(db, 'eventToEdit/'), { //when user clicks new - no events to edit so make sure eventToEdit is set to "" empty string 
+            id: "",
+            name: "",
+            description: "",
+            date: ""
+        });
+        navigate('/form', { state: { id: 1, headerText: 'Create Event', buttonText: 'Add' } }); //navigate To Form Page
+    }
+
+    //This function loads all the cards/events and their count from the database to the page and checks for edits
+    const loadEvents = () => {
+        const dbRef = ref(getDatabase(app));
+        get(child(dbRef, 'events/')).then((snapshot) => {
+            if (snapshot.exists()) {
+                let originalData = [];
+                originalData = snapshotToArray(snapshot); //CONVERT FIREBASE DATABASE SNAPSHOT/COLLECTION TO AN ARRAY                displayCards(); //intially display the cards as the order they appear in the database
+                setOriginalDataArray(originalData);
+                setFilteredDataArray(originalData);
+                setTotalCards(originalData.length);
+                setChange(true);
+            } else {
+                console.log("No data available");
+            }
+        }).catch((error) => {
+            console.error(error);
+        });
+    }
+
+    //This function is used to display the data as cards on the screen
+    const displayCards = () => {
+        let colorsArray = [];
+        //set the colors array such that, Past date:red - Current date:blue - Future date:Purple  
+        for (let i = 0; i < filteredDataArray.length; i++) {
+            if (compareDates(filteredDataArray[i].date) === 1)
+                colorsArray[i] = "#982176"; //purple
+            else if (compareDates(filteredDataArray[i].date) === 0)
+                colorsArray[i] = "#1434A4"; //blue
+            else if (compareDates(filteredDataArray[i].date) === -1)
+                colorsArray[i] = "#D22B2B"; //red
+        }
+        setCardsColorsArray(colorsArray);
+    }
+
+    //This function is used when clicking a div to edit it's event
+    const divClickHandler = (event) => {
+        let divData = event.currentTarget.textContent;
+        var myArray = divData.trim().split("  ");
+        var name = myArray[0].toUpperCase();
+
+        for (let k = 0; k < filteredDataArray.length; k++) {
+            if (name === filteredDataArray[k].name.toUpperCase()) {
+                console.log(filteredDataArray[k].name);
+                set(ref(db, 'eventToEdit/'),
+                    {
+                        id: filteredDataArray[k].id,
+                        name: filteredDataArray[k].name,
+                        description: filteredDataArray[k].description,
+                        date: filteredDataArray[k].date
+                    });
+                break;
```

Comment:
this loop used to find the element from filteredDataArray, so please use "find" javascript function

Reply:
done